File: scala/sumOfEvenSquares.scala
def pow(x: Int, y: Int): Int =
  var product = 1
  for i <- 1 to y 
  do product *= x
  return product

def ses(a: Array[Int]): Int =
  return a.filter(_ % 2 == 0).map(pow(_,2)).fold(0)(_ + _)

@main def sumOfEvenSquares() = 
  assert(ses(Array()) == 0)
  assert(ses(Array(1)) == 0)
  assert(ses(Array(1,2)) == 4)
  assert(ses(Array(1,2,3)) == 4)
  assert(ses(Array(1,2,3,4)) == 20)
  assert(ses(Array(4,6,7,2)) == 56)

File: php/sum_of_even_squares.php
<?php

function sumOfEvenSquares($a) {
    return array_sum(array_map(fn($n) => $n ** 2, 
                     array_filter($a, fn($n) => $n % 2 === 0))); 
}

assert(sumOfEvenSquares([]) === 0);
assert(sumOfEvenSquares([1]) === 0);
assert(sumOfEvenSquares([1, 2]) === 4);
assert(sumOfEvenSquares([1, 2, 3]) === 4);
assert(sumOfEvenSquares([1, 2, 3, 4]) === 20);
assert(sumOfEvenSquares([4, 6, 7, 2]) === 56);

?>